File: Backend/Microservices/Expense/EMS.Core/Entities/ScheduledEvent.cs
using EMS.Core.Entities.Common;
using EMS.Core.Enums;

namespace EMS.Core.Entities
{
    public class ScheduledEvent : BaseAuditableEntity<int>
    {
        public string UserId { get; set; } = default!;
        public string Name { get; set; } = default!;
        public string? Description { get; set; }
        public EventType Type { get; set; }
        public string? Payload { get; set; }
        public EventStatus Status { get; set; }
        public int? RecurrenceRuleId { get; set; }

        // Timing
        public DateTimeOffset InitialTrigger { get; set; }
        public DateTimeOffset? NextOccurrence { get; set; }
        public DateTimeOffset? LastOccurrence { get; set; }

        // Navigations
        public virtual IUser<string> User { get; set; } = default!;
        public virtual RecurrenceRule? RecurrenceRule { get; set; }
        public ICollection<ScheduledEventExecution> ScheduledEventExecutions { get; set; } = [];
    }
}

File: Backend/Microservices/AI/app/schemas/scheduled_event.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

from app.schemas.recurrence_rule import RecurrenceRule


class ScheduledEvent(BaseModel):
    id: int
    user_id: str
    name: str
    description: Optional[str] = None
    type: str
    payload: Optional[str] = None
    status: str
    recurrence_rule_id: Optional[int] = (
        None
    )

    # Timing
    initial_trigger: datetime
    next_occurrence: Optional[datetime] = None
    last_occurrence: Optional[datetime] = None

    created_at: Optional[datetime] = None

    recurrence_rule: Optional[RecurrenceRule] = None
